Assets/Scripts/Game/UserInterfaceWindows/DaggerfallInventoryWindow.cs

```diff
@@ -1156,9 +1157,19 @@ private void DropGoldPopup_OnGotUserInput(DaggerfallInputMessageBox sender, stri
             bool result = int.TryParse(input, out goldToDrop);
             if (!result || goldToDrop < 1 || goldToDrop > playerGold)
                 return;
-            // Check wagon weight limit
-            if (usingWagon && (remoteItems.GetWeight() + (goldToDrop / DaggerfallBankManager.gold1kg) > ItemHelper.wagonKgLimit))
-                return;
+            if (usingWagon)
+            {
+                // Check wagon weight limit
+                int wagonCanHold = ComputeCanHoldAmount(playerGold, DaggerfallBankManager.goldUnitWeightInKg, ItemHelper.wagonKgLimit - remoteItems.GetWeight());
+                if (goldToDrop > wagonCanHold)
+                {
+                    goldToDrop = wagonCanHold;
+                    DaggerfallMessageBox messageBox = new DaggerfallMessageBox(uiManager, this);
+                    messageBox.SetText(String.Format(wagonFullGold, wagonCanHold));
+                    messageBox.ClickAnywhereToClose = true;
+                    messageBox.Show();
```

Comment:
Maybe you could use this to shorten this boilerplate code for all the MsgBoxes you added:

DaggerfallUI.MessageBox(text);